ui/pages/permissions-connect/connect-page/connect-page.tsx

```diff
@@ -101,100 +109,194 @@ export const ConnectPage: React.FC<ConnectPageProps> = ({
+    (caipChainId) => allNetworksList.includes(caipChainId as CaipChainId),
   );
 
   const [showEditAccountsModal, setShowEditAccountsModal] = useState(false);
 
   // By default, if a non test network is the globally selected network. We will only show non test networks as default selected.
   const currentlySelectedNetwork = useSelector(getMultichainNetwork);
-  const currentlySelectedNetworkChainId =
-    currentlySelectedNetwork.network.chainId;
+  const currentlySelectedNetworkChainId = currentlySelectedNetwork.chainId;
   // If globally selected network is a test network, include that in the default selected networks for connection request
-  const selectedTestNetwork = testNetworks.find(
-    (network: { chainId: string }) =>
-      network.chainId === currentlySelectedNetworkChainId,
+  const selectedTestNetwork = testNetworkConfigurations.find(
+    (network: { caipChainId: CaipChainId }) =>
+      network.caipChainId === currentlySelectedNetworkChainId,
   );
 
   const defaultSelectedNetworkList = selectedTestNetwork
-    ? [...nonTestNetworks, selectedTestNetwork].map(({ chainId }) => chainId)
-    : nonTestNetworks.map(({ chainId }) => chainId);
+    ? [...nonTestNetworkConfigurations, selectedTestNetwork].map(
+        ({ caipChainId }) => caipChainId,
+      )
+    : nonTestNetworkConfigurations.map(({ caipChainId }) => caipChainId);
+
+  const defaultSelectedChainIds =
+    supportedRequestedCaipChainIds.length > 0
+      ? supportedRequestedCaipChainIds
+      : defaultSelectedNetworkList;
 
-  const allNetworksList = [...nonTestNetworks, ...testNetworks].map(
-    ({ chainId }) => chainId,
+  const [selectedChainIds, setSelectedChainIds] = useState<CaipChainId[]>(
+    defaultSelectedChainIds as CaipChainId[],
   );
 
-  const supportedRequestedChainIds = requestedChainIds.filter((chainId) =>
-    allNetworksList.includes(chainId),
+  const allAccounts = useSelector(
+    getUpdatedAndSortedAccountsWithCaipAccountId,
+  ) as MergedInternalAccountWithCaipAccountId[];
+
+  const requestedNamespaces = getAllNonWalletNamespacesFromCaip25CaveatValue(
+    requestedCaip25CaveatValue,
   );
 
-  const defaultSelectedChainIds =
-    supportedRequestedChainIds.length > 0
-      ? supportedRequestedChainIds
-      : defaultSelectedNetworkList;
+  // all accounts that match the requested namespaces
+  const supportedAccountsForRequestedNamespaces = allAccounts.filter(
+    (account) => {
+      const {
+        chain: { namespace },
+      } = parseCaipAccountId(account.caipAccountId);
+      return requestedNamespaces.includes(namespace);
+    },
+  );
 
-  const [selectedChainIds, setSelectedChainIds] = useState(
-    defaultSelectedChainIds,
+  // all requested accounts that are found in the wallet
+  const supportedRequestedAccounts = requestedCaipAccountIds.reduce(
+    (acc, account) => {
+      const supportedRequestedAccount =
+        supportedAccountsForRequestedNamespaces.find(
```

Comment:
Could these be EIP155 addresses that should be compared without casing? Just wondering if this might cause a bug in the EVM flow

Reply:
Yeah I considered this and thought it wasn't an issue, but I think it may be so I'll add that check for EVM addresses only

Reply:
The consequences here are pretty low, it would just mean a requested account (rare that specific accounts are requested at all) wouldn't get default pre-checked in case mismatch

Reply:
Yeah, but seems worth it to make sure since we specifically added this for certain dapps that asked for it 😄

Reply:
Probably not for now, but perhaps worth considering in the future to move some of this to a more shared place. So we don't have multiple implementations of checking for EIP155 etc

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/29887/commits/bc2852a2e46496b7397daef9eb4bd2dbd8cfe7d6